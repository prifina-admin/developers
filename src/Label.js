import React, { forwardRef } from "react";

import styled, { css } from "styled-components";

import { layout, space, typography, color, compose } from "styled-system";
import { default as styledProps } from "@styled-system/prop-types";
import PropTypes from "prop-types";

const systemProps = compose(layout, space, typography, color);

const nowrap = props => {
  return props.nowrap
    ? {
        whiteSpace: "nowrap",
      }
    : null;
};
const pointer = props => {
  return props.pointer ? { cursor: "pointer" } : null;
};
const hidden = props => {
  return props.hidden ? { visibility: "hidden" } : null;
};

/*
const VisuallyHidden = styled(Box)`
  border: 0px;
  clip: rect(0px, 0px, 0px, 0px);
  height: 1px;
  width: 1px;
  margin: -1px;
  padding: 0px;
  overflow: hidden;
  white-space: nowrap;
  position: absolute;
`;
*/

const LabelTheme = css`
  font-size: ${props => props.theme.textStyles[props.textStyle].fontSize};
  letter-spacing: ${props =>
    props.theme.textStyles[props.textStyle].letterSpacing};
  display: ${props => props.variation};
  width: 100%;
  margin: 0;
  color: ${props => props.theme.textStyles[props.textStyle].color};
  font-weight: ${props => props.theme.textStyles[props.textStyle].fontWeight};
  line-height: ${props => props.theme.textStyles[props.textStyle].lineHeight};
  ${systemProps};
  ${nowrap}
  ${pointer}
  ${hidden}
`;

const LabelElement = styled("label")`
  ${LabelTheme}
`;

const Label = forwardRef((props, ref) => {
  return <LabelElement {...props} ref={ref} />;
});
Label.propTypes = {
  ...styledProps.space,
  ...styledProps.layout,
  ...styledProps.typography,

  variation: PropTypes.oneOf(["block", "inline", "none"]),
  pointer: PropTypes.bool,
};
Label.defaultProps = {
  textStyle: "body",
};

Label.displayName = "Label";
Label.isLabel = true;

export default Label;
