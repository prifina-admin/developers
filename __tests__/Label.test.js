import React, {createRef} from "react";
import renderer from "react-test-renderer";
import Label from "../src/Label";
import ThemeProvider from "../src/theme/ThemeProvider";

test("Label renders correctly", () => {
    const tree = renderer.create(
        <ThemeProvider>
            <Label pointer nowrap variation={"block"}>
                Something very long text next line here{" "}
            </Label>
        </ThemeProvider>).toJSON();
    expect(tree).toMatchSnapshot();
});
