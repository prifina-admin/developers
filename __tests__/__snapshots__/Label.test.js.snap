// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label renders correctly 1`] = `
<div
  className="ThemeProvider__Base-sc-1rk52n4-0 emGfPA"
>
  <label
    className="Label__LabelElement-u5wfo1-0 eQkHuW"
  >
    Something very long text next line here
     
  </label>
</div>
`;
